xds/internal/xdsclient/transport/internal/internal.go

```diff
@@ -0,0 +1,25 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package internal contains functionality internal to the transport package.
+package internal
+
+// The following vars can be overridden by tests.
+var (
+	// GRPCNewClient creates a new gRPC Client.
+	GRPCNewClient any // func(string, ...grpc.DialOption) (*grpc.ClientConn, error)
```

Comment:
nit:

```suggestion
	GRPCNewClientForTesting any // func(string, ...grpc.DialOption) (*grpc.ClientConn, error)
```

also not entirely clear why we are stubbing gRPC.NewClient like this vs how it was before. Is this in anticipation to the new xds module we are planning?

Reply:
The reason is for moving the test in `xds/internal/xdsclient/transport/transport_test.go` to a test only package, thereby avoiding the possibility of inadvertently using some unexported functionality from the package.

This is a pattern that was suggested by Doug, and I have been using it wherever applicable, i.e, instead of making a test live in the same package as that of the code being tested **only** for the reason of being able to override some functionality, we can have an `internal` package in the same directory that contains vars that can be overridden at test time, and yet the test can live in its own package.

And we generally don't name these vars with a `ForTesting` suffix, because it will be used in the package's code as well. So, suddenly seeing something in non-test code with a `ForTesting` suffix can throw us off.